hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/HeaderPreprocessor.java

```diff
@@ -17,39 +17,58 @@
  */
 package org.apache.hadoop.ozone.s3;
 
+import javax.annotation.Priority;
 import javax.ws.rs.container.ContainerRequestContext;
 import javax.ws.rs.container.ContainerRequestFilter;
 import javax.ws.rs.container.PreMatching;
 import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.MultivaluedMap;
 import javax.ws.rs.ext.Provider;
 import java.io.IOException;
 
 /**
  * Filter to adjust request headers for compatible reasons.
+ *
+ * It should be executed AFTER signature check (VirtualHostStyleFilter).
  */
-
 @Provider
 @PreMatching
+@Priority(150)
```

Comment:
Is this done, to perform this Preprocessing after VirtualHostStyleFilter processing.
Can we define these constants and use them, instead of hardcoded values.

Reply:
> Is this priority setting is done, to perform this Preprocessing after VirtualHostStyleFilter processing. (Is there any reason to do in this order?)

Yes, the Content-Type headers can be part of the headers which are part of the signature. Therefore we need to check the signature first and modify the Content-Type header after.

 >  Can we define these constants and use them, instead of hardcoded values.

Well, it's hardcoded value anyway and used only in this one place. But I improved it with constants to express the dependencies:

```
@Priority(VirtualHostStyleFilter.PRIORITY
    + S3GatewayHttpServer.FILTER_PRIORITY_DO_AFTER)
public class HeaderPreprocessor implements ContainerRequestFilter {
```

Reply:
This approach LGTM, clear to understand.